File: src/main/java/org/lvtinger/paladin/web/AccountController.java
package org.lvtinger.paladin.web;

import org.apache.commons.lang3.StringUtils;
import org.lvtinger.paladin.account.api.AccountService;
import org.lvtinger.paladin.account.api.SessionService;
import org.lvtinger.paladin.value.Result;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import javax.annotation.Resource;

@RestController
@RequestMapping(value = "/account")
public class AccountController {
    @Resource
    private AccountService accountService;
    @Resource
    private SessionService sessionService;

    @RequestMapping(value = "/register", method = RequestMethod.POST)
    public Result<String> register(String username, String password, String device) {
        if (StringUtils.isAnyBlank(username, password, device)) {
            return Result.<String>warning().touchMessage("参数错误");
        }
        return accountService.register(username, password, device);
    }

    @RequestMapping(value = "/login", method = RequestMethod.POST)
    public Result<String> login(String username, String password, String device) {
        if (StringUtils.isAnyBlank(username, password, device)) {
            return Result.<String>warning().touchMessage("参数错误");
        }
        return accountService.login(username, password, device);
    }

    @RequestMapping(value = "/logout", method = RequestMethod.POST)
    public Result<Boolean> logout(String token) {
        if (StringUtils.isEmpty(token)) {
            return Result.success(false);
        }
        if (sessionService.exp(token).getCode() == Result.SUCCESS) {
            return Result.success(true);
        } else {
            return Result.success(false);
        }
    }
}

File: src/main/kotlin/org/risesun/paladin/controller/AccountController.kt
package org.risesun.paladin.controller

import org.risesun.paladin.core.controller.AbstractController
import org.risesun.paladin.provider.api.AccountService
import org.risesun.paladin.value.AccountRequestValue
import org.risesun.paladin.value.ResponseMessage
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RestController

@RequestMapping(value = ["/account"])
@RestController
open class AccountController : AbstractController() {

    @Autowired
    private lateinit var accountService: AccountService

    @RequestMapping(value = ["/login.page"], method = [RequestMethod.POST])
    fun login(@RequestBody parameter: AccountRequestValue): ResponseMessage<Any?> {
        val message = parameter.validate()
        if (message.success != true) {
            return message as ResponseMessage<Any?>
        }

        val login = accountService.login(parameter.username!!, parameter.password!!)

        return login as ResponseMessage<Any?>
    }

    @RequestMapping(value = ["/register.page"], method = [RequestMethod.POST])
    fun register(@RequestBody parameter: AccountRequestValue): ResponseMessage<Any?> {
        val message = parameter.validate()
        if (message.success != true) {
            return message as ResponseMessage<Any?>
        }

        val register = accountService.register(parameter.username!!, parameter.password!!)
        return register as ResponseMessage<Any?>
    }
}